src/BenchmarksArtisanCommand.php

```diff
@@ -32,10 +43,56 @@ public function handle(): void
         }
 
         $this->startBenchmark();
-        $this->call($commandToBenchmark);
+        $this->call($commandToBenchmark, $this->getArgumentsToPassThough());
         $this->endBenchmark();
     }
 
+    protected function getArgumentsToPassThough(): array
+    {
+        $commandToBenchmark = $this->argument('signature');
+
+        if (! $commandToBenchmark) {
+            return [];
+        }
+
+        $command = Artisan::all()[$commandToBenchmark] ?? null;
+
+        if (! $command) {
+            throw new CommandNotFoundException(
+                \sprintf('Command "%s" does not exist.', $commandToBenchmark)
+            );
+        }
+
+        $booleanOptions = array_keys(array_filter(
+            $command->getDefinition()->getOptions(),
+            fn (InputOption $option) => ! $option->acceptValue(),
+        ));
+
+        $tokens = collect((new StringInput($this->input))->getRawTokens())
+            ->reject(fn (string $token): string => str_starts_with($token, '--tableToWatch='))
+            ->map(function (string $token) use ($booleanOptions): string {
+                $optionName = Str::match('/^--([^=]+)=/', $token);
+
+                if (in_array($optionName, $booleanOptions)) {
+                    return '--' . $optionName;
+                }
+
+                return $token;
+            })
+            ->slice(1)
+            ->toArray();
+
+        $input = new ArgvInput($tokens, $command->getDefinition());
+
+        return [
+            ...$input->getArguments(),
+            ...Arr::mapWithKeys(
+                $input->getOptions(),
+                fn (array|string|null $option, string $key): array => [Str::start($key, '--') => $option]
+            ),
+        ];
```

Comment:
If there is a better way of doing this, I would love to hear it! I fell down a bit of a rabbit hole :D

Reply:
hehe, this is indeed a tricky one. Sounds confusing with the different options for the different commands 🤔

What about a new syntax when there is a need for options on the command to check:

`php artisan benchmark {import:users --myFlag} --tableToWatch=users`

I think it might be easier to parse this into commands and options/arguments, but I haven't tested it.

Besides, we could get away from benchmark options and provide the features through prompts, but this also feels a bit limited. What do you think?

Reply:
passing the whole command as an argument is a great idea! I'll play around with that a bit later. I guess it would need to be wrapped in quotes?

Reply:
an alternative solution would be to do this the other way around, have `--benchmark` be available as a global option for all commands. But not sure that can be done via a package or would involve modifying internals